### PR TITLE
Refactor http binding to support https

### PR DESCRIPTION
**Pull request by pravinpushkar**
# Description

_Please explain the changes you've made_

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #2765 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: bindings/http/http.go [hunk 1]
```diff
@@ -98,7 +101,7 @@ func (h *HTTPSource) Init(_ context.Context, meta bindings.Metadata) error {
 		Dial:                dialer.Dial,
 		TLSHandshakeTimeout: 5 * time.Second,
 	}
-	if tlsConfig != nil && len(tlsConfig.Certificates) > 0 && tlsConfig.RootCAs != nil {
+	if tlsConfig != nil {
 		netTransport.TLSClientConfig = tlsConfig
 	}
```
Comment (ItalyPaleAle) [inline bindings/http/http.go:106]: Nit:


```suggestion
		TLSHandshakeTimeout: 5 * time.Second,
		TLSClientConfig:     tlsConfig,
	}
```

File: bindings/http/http.go [hunk 2]
```diff
@@ -116,37 +119,46 @@ func (h *HTTPSource) Init(_ context.Context, meta bindings.Metadata) error {
...
+		tlsConfig = &tls.Config{MinVersion: tls.VersionTLS12}
 	}
+	tlsConfig.Certificates = []tls.Certificate{cert}
+	return nil
+}
+
+// Add Root CA cert to the pool of trusted certificates.
+// This is required for the client to trust the server certificate in case of HTTPS connection.
+func (h *HTTPSource) addRootCAToCertPool() (*tls.Config, error) {
+	var tlsConfig *tls.Config
```
Comment (ItalyPaleAle) [inline bindings/http/http.go:146]: Nit: Consider short-circuiting instead (for readability):

```go
func (h *HTTPSource) addRootCAToCertPool() (*tls.Config, error) {
    if h.metadata.MTLSRootCA == "" {
        return nil, nil
    }

    // Do the rest
```

File: bindings/http/http.go [hunk 3]
```diff
@@ -116,37 +119,46 @@ func (h *HTTPSource) Init(_ context.Context, meta bindings.Metadata) error {
...
 	if h.metadata.MTLSRootCA != "" {
 		caCertBytes, err := h.getPemBytes(MTLSRootCA, h.metadata.MTLSRootCA)
 		if err != nil {
 			return nil, err
 		}
+
 		caCertPool := x509.NewCertPool()
-		ok := caCertPool.AppendCertsFromPEM(caCertBytes)
-		if !ok {
+		if ok := caCertPool.AppendCertsFromPEM(caCertBytes); !ok {
```
Comment (ItalyPaleAle) [inline bindings/http/http.go:154]:
```suggestion
		if !caCertPool.AppendCertsFromPEM(caCertBytes) {
```

**Review by ItalyPaleAle — COMMENTED** [review]
Is this PR ready for review? I see title still says [WIP]

**Comment by pravinpushkar** [comment]
> Is this PR ready for review? I see title still says [WIP]

It is ready for review, forgot to remove the WIP. I will address the comments shortly.
Thanks for taking a look @ItalyPaleAle

**Comment by pravinpushkar** [comment]
@ItalyPaleAle  Review comments are addressed.

**Review by ItalyPaleAle — APPROVED** [review]

**Pull request merged**